### ISSUE TITLE
TypeError: unrecognized operator 'Shape'

### ISSUE THREAD

**Issue by astu9880** [Microsoft/onnxjs#84]
onnxjs provides no support for Shape operator. I couldn't find anything about it in the docs as well. Which onnx version does onnx.js support?

**Comment by hariharans29** [comment]
Hi,

Thanks for your interest in ONNX.js. Currently, ONNX.js supports [ONNX V1.2.3 - Opset V7](https://github.com/onnx/onnx/blob/rel-1.2.3/docs/Operators.md). Please keep in mind- currently ONNX.js only supports only a subset of all operators as listed [here](docs/operators.md). We are adding support for more operators on a regular basis.

If it's possible, could you please share your ONNX model for us to investigate feasibility for full ONNX.js support ?

Thanks

**Comment by rcmalli** [comment]
This is reproducible by using `resnet50` model by using Pytorch 1.0+. Shape operator is used by the converter.

```python
import torch.onnx
import torchvision

dummy_input = torch.randn(1, 3, 224, 224)
model_resnet = torchvision.models.resnet50(pretrained=True)
torch.onnx.export(model_resnet, dummy_input, "resnet50.onnx",verbose=True)
```

**Comment by NTT123** [comment]
As described in the source code https://github.com/pytorch/vision/blob/1f8f7ea71c8807408a7860b8ae9ca63177975b64/torchvision/models/resnet.py#L161
You can override the `forward` function to remove `x.size(0)` from the model, e.g., x.size(0) = 1 if your batch size = 1.

**Comment by astu9880** [comment]
@NTT123 We tried removing x.size() from our code. We are using x.view(). Could that be a reason for the fail?

**Comment by euler16** [comment]
@NTT123 and @hariharans29 is there a way through which we can avoid ONNX Shape operator by manually changing the ONNX IR as suggested [here](https://discuss.pytorch.org/t/export-onnx-with-static-shape-instead-of-onnx-shape-operator/27626)?

**Comment by NTT123** [comment]
@astu9880, for Resnet Model, I recommend a hack by replacing:
`x = x.view(x.size(0), -1)` by two lines:

` x = x.view(-1)`
` x = x.unsqueeze(0)`

this assumes your batchsize = 1

**Comment by SirCameron** [comment]
Hi

I'm using PyTorch to export a custom CNN model. 
A piece of the verbose export shows:
`%11 : Long() = onnx::Constant[value={0}](), scope: Model
  %12 : Tensor = onnx::Shape(%0), scope: Model
  %13 : Long() = onnx::Gather[axis=0](%12, %11), scope: Model
  %14 : Long() = onnx::Constant[value={1}](), scope: Model
  %15 : Tensor = onnx::Shape(%0), scope: Model
  %16 : Long() = onnx::Gather[axis=0](%15, %14), scope: Model
  %17 : Long() = onnx::Constant[value={2}](), scope: Model
  %18 : Tensor = onnx::Shape(%0), scope: Model`

Too many internal Shape operators to override. Possibly the internals of a convolution layer.

Any updates on the Shape operator in onnxjs?

**Comment by astu9880** [comment]
@hariharans29 @NTT123 
The following is the link to my onnx model.
https://drive.google.com/open?id=1EVNCW_cNGx2mw2z_aXpGiSdnnElotQ6_
Request you to add functionality in onnx.js for the operators used as I am not able to successfully load the model in the latest version of onnxjs.

**Comment by fs-eire** [comment]
@astu9880 I would like to take a look at the model when I got time. The link is 404 now

**Comment by astu9880** [comment]
@fs-eire Thanks. I have updated the link.

**Comment by mike-holcomb** [comment]
I was running into the same issues with ONNX files based off of fine-tuned torchvision models.  It looks like the heads of the official torchvision models were recently changed in the master branch (https://github.com/pytorch/vision/pull/1134) to remove this call to `size()` in favor of `flatten()` which appears to be supported by onnx.js.

**Comment by Krutsch** [comment]
This is true, but the onnx Model stills fails with "cannot resolve operator 'Shape' with opsets: ai.onnx v9"

**Comment by davidpfahler** [comment]
@mike-holcomb Unfortunately, I am still seeing this error when using resnet50 from torchvision 0.4.0

**Comment by AiueoABC** [comment]
I thought I can just skip "Shape" layer if that value is constant, and managed to avoid getting error "TypeError: unrecognized operator 'Shape'".
But this fails when I try to load for inference.

For this reproduction, I put what I did;
```
import onnx
from onnx import helper
model =  './test_learn.onnx'
onnx_model = onnx.load(model)
for i,nodes in enumerate(onnx_model.graph.node):
    print("=====")
    print(nodes)

# After finding Shape layer number, I tried to see what the value is
intermediate_layer_value_info = helper.ValueInfoProto()
intermediate_layer_value_info.name = "356"  # intermediate_tensor_name(output name of the layer, using "Netron" will help finding this.)
onnx_model.graph.output.append(intermediate_layer_value_info)
onnx.save(onnx_model, './dbg_test_learn.onnx')
'''
I used above with onnxruntime, and findout array([1,1024,1,1]) is the constant value.
After finding this, I replaced "Shape" layer with constant one.
'''
onnx_model = onnx.load(model)
node_number = 89
node = onnx_model.graph.node[node_number]
tensors = onnx.helper.make_tensor(
                name='value1',
                data_type=onnx.TensorProto.FLOAT,
                dims=(4,),
                vals=[1,1024,1,1]
            )
new_node = onnx.helper.make_node(
    op_type = "Constant",
    name='Constant_89',
    inputs=[],
    outputs=['356'],
    value=tensors
)
onnx_model.graph.node.remove(node)
onnx_model.graph.node.insert(node_number, new_node)
onnx.save(onnx_model, './dbg2_test_learn.onnx')
```
The model I got here never works on both onnxruntime and onnxjs.
Is there anyone who can help me with this?
